### PR TITLE
Improve unit test coverage

### PR DESCRIPTION
**Pull request by mckennagalvin**
Increase coverage for `reply` and `request_handler_static`

### CODE REVIEW

File: test/request_handler_static_test.cpp [hunk 1]
```diff
@@ -9,46 +9,68 @@
...
+	HandleStaticRequest();
+	EXPECT_EQ(rep.status, http::server::reply::not_found);
 }
 
-TEST_F(RequestHandlerStatic, IndexTest) {
-	server_options_.static_files_map["/static1"] = "www";
-	http::server::request_handler_static handler(&server_options_);
-	req.uri = "/static1/";
-	handler.handle_request(req, rep);
-	EXPECT_EQ(rep.content, "<!DOCTYPE html>\n<html>\n<body>\n\n<h1>This is default.html</h1>\n\n<p>The webserver successfully served this static file YAY</p>\n\n</body>\n</html>\n");
```
Comment (mattdrag) [inline test/request_handler_static_test.cpp:46]: Is there any reason that you took this test out?

File: test/request_handler_static_test.cpp [hunk 2]
```diff
@@ -9,46 +9,68 @@
...
+
+TEST_F(RequestHandlerStatic, BadRequest) {
 	req.uri = "thisWontWork.jpg";
-	handler.handle_request(req, rep);
-	EXPECT_EQ(rep.content, "<html>"
-  "<head><title>Bad Request</title></head>"
-  "<body><h1>400 Bad Request</h1></body>"
-  "</html>");
+	HandleStaticRequest();
+	EXPECT_EQ(rep.status, http::server::reply::bad_request);
```
Comment (mattdrag) [inline test/request_handler_static_test.cpp:40]: Nice change, it is a lot cleaner to use the reply:: enums for this

**Comment by mattdrag** [comment]
Nice! looks like some thorough test cases

File: test/request_handler_static_test.cpp (hunk 1 above)
Comment (mckennagalvin) [reply test/request_handler_static_test.cpp:46]: It's still there! Just named "SimpleRequest"

Comment (mattdrag) [reply test/request_handler_static_test.cpp:46]: Whoops! Didn't see that haha

**Pull request merged**